doc-requirements.txt

```diff
@@ -0,0 +1,4 @@
+sphinx >= 4.0
```

Comment:
Nothing else uses this file so it is likely to get out of date. Can we just list the requirements by pypi package name in the document instead? It is not a long or convoluted list.

I don't want to force devs to use conda, but there is a conda environment file in `sphinx-doc` that readthedocs uses when building the docs, so that one is tested.